### ISSUE TITLE
urlunsplit for itms-services scheme returns invalid url

### ISSUE THREAD

**Issue by AndyQ** [python/cpython#104139]
Relating to a Werkzueg issue (https://github.com/pallets/werkzeug/issues/2691), when parsing an iOS App install url e.g.
itms-services:action=download-manifest&url=https://theacmeinc.com/abcdefeg, urlunpslit returns an invalid url.

e.g.

```from urllib.parse import urlparse, urlunsplit
vals = urlparse( "itms-services://?action=download-manifest&url=https://theacmeinc.com/abcdefeg" )
print(vals)

newURL = urlunsplit((vals.scheme, vals.netloc, vals.path, vals.query, vals.params))
print( newURL )
```

prints:
```
ParseResult(scheme='itms-services', netloc='', path='', params='', query='action=download-manifest&url=https://theacmeinc.com/abcdefeg', fragment='')

itms-services:?action=download-manifest&url=https://theacmeinc.com/abcdefeg
```

Note the newURL is missing the // after the itms-services scheme.

This scheme is used to install ad-hoc and enterprise iOS apps.

# Your environment
Tested on Apple M1 Max - 13.4 Beta (22F5049e)
Python:  3.10.10 

For more details on the scheme here is a link to the Apple documentation (look for the "Use a website to distribute the app" section).
https://support.apple.com/en-gb/guide/deployment/depce7cefc4d/web

### Linked PRs
* gh-104312

**Comment by davidism** [comment]
I don't think this scheme is standard. https://url.spec.whatwg.org/#url-serializing says "If url's `host` is non-null: Append "//" to output." This `itms-services` URL doesn't have a host, so it shouldn't require a `//`. I think there's some other special casing for schemes with https://github.com/python/cpython/blob/9f9e001ab2ab67acdb2e0383a25ab4c164608a47/Lib/urllib/parse.py#L44-L74, so "don't add `//`" could be added to those checks.

**Comment by gpshead** [comment]
I agree with @davidism that the WhatWG URL spec does not require // when there is no pathname.

Regardless, behavior wise this seems to match our existing `uses_netloc` special case so we can just add it to that list in Lib/urllib/parse.py and add test coverage in Lib/test/test_urlparse.py.  I made a PR.

Workaround for this to "work" on existing Pythons:

```python
if "itms-services" not in urllib.parse.uses_netloc:
    urllib.parse.uses_netloc.append("itms-services")
```

I'm calling this a feature as code will have to deal with Python's that do not list it as such for a long time anyways via a code snippet like that, so backporting doesn't seem consistently helpful.

**Comment by gpshead** [comment]
fixed in 3.12.
